delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                        'storage_id': storage_id,
+                        'native_disk_id': disk_map['DeviceID'],
+                        'serial_number': serial_number,
+                        'manufacturer': vendor,
+                        'model': model,
+                        'firmware': version,
+                        'speed': None,
+                        'capacity': int(self.get_size(size)),
+                        'status': status,
+                        'physical_type': None,
+                        'logical_type': None,
+                        'native_disk_group_id': pool_id,
+                        'location': disk_map['Serialnumber'],
+                    }
+                    disks_list.append(disk_model)
+            return disks_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get disk from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_pool_size(self):
+        size_info = self.ssh_do_exec([constant.POOL_SIZE_COMMAND])
+        size_array = size_info.split("\r\n")
+        size_map = {}
+        pool_name = None
+        count = 0
+        for size in size_array:
+            if 'Span ' in size:
+                pool_name = size.split()[-1].replace(':', '')
+                size_map[pool_name] = 0
+                count = 0
+            if '[Free space]' in size:
+                free_array = size.split()
+                if len(free_array) > 2:
+                    count += 1
+                    free_size = free_array[0].replace('GiB', 'GB')
+                    size_map[pool_name] += self.get_size(free_size)
+        return size_map
+
+    def get_pool(self, storage_id):
+        try:
+            pool_info = self.ssh_do_exec([constant.POOL_INFO_COMMAND])
+            pool_list = []
+            pool_array = self.get_table_data(pool_info)
+            size_map = self.get_pool_size()
+            for pool in pool_array:
+                value_array = pool.split()
+                if len(value_array) == 6:
+                    total_capacity = \
+                        self.get_size(value_array[3] + "GB")
+                    free_capacity = \
+                        size_map.get(value_array[0], total_capacity)
+                    status = constants.StoragePoolStatus.NORMAL \
+                        if value_array[1] == 'Yes' \
+                        else constants.StoragePoolStatus.ABNORMAL
+                    pool_model = {
+                        'name': value_array[0],
+                        'storage_id': storage_id,
+                        'native_storage_pool_id': value_array[0],
+                        'description': None,
```

Comment:
If info cannot be attained from storage, it can be deleted.

Reply:
has been modify